application/shared-kernel/ApiCore/OpenApi/OpenApiDocumentGeneratorSettingsExtensions.cs

```diff
@@ -0,0 +1,26 @@
+using System.Text.Json;
+using Microsoft.AspNetCore.Http.Json;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using NJsonSchema.Generation;
+using NSwag.Generation.AspNetCore;
+
+namespace PlatformPlatform.SharedKernel.ApiCore.OpenApi;
+
+public static class OpenApiDocumentGeneratorSettingsExtensions
+{
+    /// <summary>
+    ///     Configures the schema settings for the OpenAPI document generator
+    /// </summary>
+    public static void ConfigureSchemaSettings(
+        this AspNetCoreOpenApiDocumentGeneratorSettings settings,
+        IServiceProvider serviceProvider
+    )
+    {
+        if (settings.SchemaSettings is not SystemTextJsonSchemaGeneratorSettings schemaSettings) return;
```

Comment:
I would inline this entire method, and then delete this check, as settings.SchemaSettings would then always be the same type.

Reply:
Seems like this check is required as SchemaSettings is type `JsonSchemaGeneratorSettings`